File: exchange-desktop/src/main/java/com/maarouf/exchange/api/model/InsightsData.java
package com.maarouf.exchange.api.model;

import com.google.gson.annotations.SerializedName;

import java.util.Map;

public class InsightsData {
    @SerializedName("lbp_to_usd_avg")
    public Map<String, Float> lbpToUsdAverage;

    @SerializedName("usd_to_lbp_avg")
    public Map<String, Float> usdToLbpAverage;

    @SerializedName("usd_to_lbp_open")
    public Map<String, Float> usdToLbpOpen;

    @SerializedName("lbp_to_usd_open")
    public Map<String, Float> lbpToUsdOpen;

    @SerializedName("usd_to_lbp_close")
    public Map<String, Float> usdToLbpClose;

    @SerializedName("lbp_to_usd_close")
    public Map<String, Float> lbpToUsdClose;

    @SerializedName("volume_in_trxs")
    public  Map<String, Long> volumeInTransactions;

    @SerializedName("volume_in_usd")
    public Map<String, Float> volumeInUsd;

}

File: exchange-android/app/src/main/java/com/maarouf/currencyexchange/api/model/InsightsData.kt
package com.maarouf.currencyexchange.api.model

import com.google.gson.annotations.SerializedName

class InsightsData {
    @SerializedName("lbp_to_usd_avg")
    var lbpToUsdAverage: Map<String, Float>? = null

    @SerializedName("usd_to_lbp_avg")
    var usdToLbpAverage: Map<String, Float>? = null

    @SerializedName("usd_to_lbp_open")
    var usdToLbpOpen: Map<String, Float>? = null

    @SerializedName("lbp_to_usd_open")
    var lbpToUsdOpen: Map<String, Float>? = null

    @SerializedName("usd_to_lbp_close")
    var usdToLbpClose: Map<String, Float>? = null

    @SerializedName("lbp_to_usd_close")
    var lbpToUsdClose: Map<String, Float>? = null

    @SerializedName("volume_in_trxs")
    var volumeInTransactions: Map<String, Long>? = null

    @SerializedName("volume_in_usd")
    var volumeInUsd: Map<String, Float>? = null
}